pages/admin/users.js

```diff
@@ -0,0 +1,116 @@
+import { authOptions } from "../api/auth/[...nextauth]";
+import { getServerSession } from "next-auth/next";
+import { clientEnv } from "@config/schemas/clientSchema";
+
+import logger from "@config/logger";
+import Page from "@components/Page";
+import PageHead from "@components/PageHead";
+import { getUsers } from "../api/admin/users";
+
+import { serverEnv } from "@config/schemas/serverSchema";
+import Navigation from "@components/admin/Navigation";
+import { ChevronRightIcon } from "@heroicons/react/20/solid";
+import Image from "next/image";
+
+export async function getServerSideProps(context) {
+  const session = await getServerSession(context.req, context.res, authOptions);
+  if (!session) {
+    return {
+      redirect: {
+        destination: "/auth/signin",
+        permanent: false,
+      },
+    };
+  }
+
+  const username = session.username;
+
+  if (!serverEnv.ADMIN_USERS.includes(username)) {
+    return {
+      redirect: {
+        destination: "/404",
+        permanent: false,
+      },
+    };
+  }
+
+  let users = [];
+  try {
+    users = await getUsers();
+  } catch (e) {
+    logger.error(e, "get users failed");
+  }
+  return {
+    props: {
+      username,
+      users,
+      BASE_URL: clientEnv.NEXT_PUBLIC_BASE_URL,
+    },
+  };
+}
+
+export default function USERS({ users }) {
+  return (
+    <>
+      <PageHead
+        title="LinkFree admin users"
+        description="Overview for LinkFree admins"
+      />
+      <Page>
+        <Navigation />
+        <h1 className="text-4xl mb-4 font-bold">Users</h1>
+        <ul role="list" className="divide-y divide-gray-100">
+          {users.map((person) => (
```

Comment:
I think using the singular of `users` here would be more clear

```suggestion
          {users.map((user) => (
```